Examples/Tests/PythonWrappers/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
+from pywarpx import picmi
+
+# Number of time steps
+max_steps = 100
+
+# Grid
+nx = 128
+nz = 128
+
+# Domain
+xmin = 0.e-6
+zmin = 0.e-6
+xmax = 50.e-6
+zmax = 50.e-6
+
+# Cell size
+dx = (xmax - xmin) / nx
+dz = (zmax - zmin) / nz
+
+# Domain decomposition
+max_grid_size_x = 64
+max_grid_size_z = 64
+
+# PML
+nxpml = 10
+nzpml = 10
+field_boundary = ['open', 'open']
+
+# Spectral order
+nox = 8
+noz = 8
+
+# Guard cells
+nxg = 8
+nzg = 8
+
+# Initialize grid
+grid = picmi.Cartesian2DGrid(number_of_cells = [nx,nz],
+                             lower_bound = [xmin,zmin],
+                             upper_bound = [xmax,zmax],
+                             lower_boundary_conditions = field_boundary,
+                             upper_boundary_conditions = field_boundary,
+                             guard_cells = [nxg,nzg],
+                             moving_window_velocity = [0.,0.,0],
+                             warpx_max_grid_size_x = max_grid_size_x,
+                             warpx_max_grid_size_y = max_grid_size_z)
+
+# Initialize field solver
+solver = picmi.ElectromagneticSolver(grid=grid, cfl=0.95, method='PSATD',
+                                     stencil_order = [nox,noz],
+                                     divE_cleaning = 1,
+                                     divB_cleaning = 1,
+                                     pml_divE_cleaning = 1,
+                                     pml_divB_cleaning = 1,
+                                     warpx_psatd_update_with_rho = True)
+
+# Initialize diagnostics
+diag_field_list = ["E", "B"]
+field_diag = picmi.FieldDiagnostic(name = 'diag1',
+                                   grid = grid,
+                                   period = 10,
+                                   write_dir = '.',
+                                   warpx_file_prefix = 'Python_wrappers_plt',
+                                   data_list = diag_field_list)
+
+# Initialize simulation
+sim = picmi.Simulation(solver = solver,
+                       max_steps = max_steps,
+                       verbose = 1,
+                       particle_shape = 'cubic',
+                       warpx_current_deposition_algo = 'direct',
+                       warpx_particle_pusher_algo = 'boris',
+                       warpx_field_gathering_algo = 'energy-conserving',
+                       warpx_use_filter = 1)
+
+# Add diagnostics to simulation
+sim.add_diagnostic(field_diag)
+
+# Write input file to run with compiled version
+sim.write_input_file(file_name = 'inputs_2d')
+
+# Whether to include guard cells in data returned by Python wrappers
+include_ghosts = 1
+
+# Compute min and max of fields data
+def compute_minmax(data):
+    vmin = -max(abs(data.min()), abs(data.max()))
```

Comment:
This could be done more succinctly.

> vmax = np.abs(data).max()
vmin = -vmax

Reply:
Thank you for the suggestion, fixed in https://github.com/ECP-WarpX/WarpX/pull/2576/commits/66ec869749ffde6c8ed047729d77f002c1d5a21b (along with other things).